### PR TITLE
Add customer set password url to Site settings

### LINKED ISSUE #4830: Configure storefront URL through the API
In the dashboard, we have a view for creating customers, which can automatically send them a link to set up their passwords. The issue is that the dashboard doesn't know what is the storefront's URL, and in particular, what is the address of a view to set the password. 

I suggest that we add a field for this in the `SiteSettings` model and allow for managing it thorough the API. The field could be called something like `customerPasswordResetURL`. 

If we need to support multiple storefronts or allow for more configurable links (e.g. storefront's product details page URL etc. we will extend this in the future).

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #4830 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 90332b3de32fac7c739e15c2008e1796cbc188e1 (mirumee/saleor @ 4830/configure_customer_password_reset_url_through_API)
Base comparison is fe68e5e59b244fc86a0bc8e04eabb0f00a611149.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         14	         14	              0
  complete checkout                          	          6	          6	              0
  create checkout                            	         48	         48	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	         13	         13	              2

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/account/mutations/base.py [hunk 1]
```diff
@@ -337,6 +346,8 @@ def clean_input(cls, info, instance, data):
             )
             cleaned_input[BILLING_ADDRESS_FIELD] = billing_address
 
+        # DEPRECATED: We shoud remove this confition whe drop `send_password_email`
```
Comment (maarcingebala) [inline saleor/graphql/account/mutations/base.py]: Something's wrong with the grammar in this comment. Check this sentence in Grammarly.

File: saleor/graphql/shop/mutations.py [hunk 2]
```diff
@@ -42,6 +43,9 @@ class ShopSettingsInput(graphene.InputObjectType):
     default_mail_sender_address = graphene.String(
         description="Default email sender's address."
     )
+    customer_set_password_url = graphene.String(
+        description="Storefront url address in which the client sets the password."
```
Comment (maarcingebala) [inline saleor/graphql/shop/mutations.py]:
```suggestion
        description="URL of a view where customers can set their password."
```

File: saleor/graphql/shop/types.py [hunk 3]
```diff
@@ -149,6 +149,10 @@ class Shop(graphene.ObjectType):
     company_address = graphene.Field(
         Address, description="Company address.", required=False
     )
+    customer_set_password_url = graphene.String(
+        description="storefront url address in which the client sets the password.",
```
Comment (maarcingebala) [inline saleor/graphql/shop/types.py]:
```suggestion
        description="URL of a view where customers can set their password.",
```

**Pull request merged**